scipy/interpolate/tests/test_bsplines.py

```diff
@@ -1346,6 +1346,19 @@ def test_deriv_spec(self):
         with assert_raises(ValueError):
             make_interp_spline(x, y, bc_type=(l, r))
 
+    def test_deriv_order_too_large(self):
+        x = np.arange(7)
+        y = x**2
+        l, r = [(1, 0)], [(6, 0)]    # 6th derivative = 0 at x[-1] for k=3
+        with assert_raises(ValueError):
+            # cannot fix 6th derivative at x[-1]: does not segfault
+            make_interp_spline(x, y, bc_type=(l, r))
+
+        l, r = [(1, 0)], [(-6, 0)]    # derivative order < 0 at x[-1]
+        with assert_raises(ValueError):
+            # does not segfault
+            make_interp_spline(x, y, bc_type=(l, r))
```

Comment:
Sorry I should have put this with my original comment but could we not make one of the bad BCs on the left so we get coverage of both branches? Also if there is an edit to be made can we add `match=.....` too